include/ring-core/target.h

```diff
@@ -42,9 +42,9 @@
 #elif !(defined(__ORDER_LITTLE_ENDIAN__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)) && \
       !(defined(__ORDER_BIG_ENDIAN__) && (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__))
 #error "Unsupported endianness"
-#elif defined(__LP64__)
+#elif defined(__LP64__) || defined(__loongarch64) || defined(__loongarch_lp64)
 #define OPENSSL_64_BIT
-#elif defined(__ILP32__)
+#elif defined(__ILP32__) || defined(__loongarch__)
```

Comment:
Wouldn't this addition be better placed ...

Reply:
> `__loongarch__` is defined for 64-bit systems as well, so this expression doesn't seem correct.

Yes, but `__LP64__` will be set for 64-bit systems, right? Otherwise the logic for the other 32-bit archs wouldn't work either. Keep in mind that the thing we're really working around is that these old versions of GCC didn't set `__ILP32__` for many 32-bit architectures.